sktime/data/util.py

```diff
@@ -102,8 +114,41 @@ def timeshifted_split(inputs, lagtime: int, chunksize: int = 1000, stride: int =
             while t < len(data):
                 if t == len(data_lagged):
                     break
+                x = data[ix[t:min(t + chunksize, len(data))]]
                 if lagtime > 0:
-                    yield data[t:min(t+chunksize, len(data))], data_lagged[t:min(t+chunksize, len(data_lagged))]
+                    x_lagged = data_lagged[ix[t:min(t + chunksize, len(data_lagged))]]
+                    yield x, x_lagged
                 else:
-                    yield data[t:min(t+chunksize, len(data))]
+                    yield x
                 t += chunksize
+
+
+class TimeSeriesDataset(object):
+    r"""
+    High-level container for time-series data. This is intended to be used together with pytorch data tools, i.e.,
+    data loaders and other utilities.
+    """
+
+    def __init__(self, data, lagtime: int = 0):
```

Comment:
i'm not sure if this is a good idea, but would you want to give an option to stride the dataset here?

Reply:
i think later this might be a good addition but for now I would prefer to leave it at this basic functionality